docs/proposals/nginx-extensions.md

```diff
@@ -0,0 +1,964 @@
+
+### Features with Active Gateway API Enhancement Proposals
+
+The status field in the table describes the status of the GEP using the following terms:
+
+- Experimental: GEP is implemented and in the experimental channel.
+- Provisional: GEP is written but has yet to be implemented.
+- Open: An issue exists, but a GEP still needs to be written.
+
+| Functionality                                 | Status       | GEP/Issue                                                                |
+|-----------------------------------------------|--------------|--------------------------------------------------------------------------|
+| Per-request timeouts                          | Experimental | [GEP-1742](https://gateway-api.sigs.k8s.io/geps/gep-1742/)               |
+| Retries. This is different from NGINX retries | Open         | [GEP-1731](https://github.com/kubernetes-sigs/gateway-api/issues/1731)   |
+| Session persistence                           | Provisional  | [GEP-1619](https://gateway-api.sigs.k8s.io/geps/gep-1619/)               |
+| Rate-limiting                                 | Open         | [Issue 326](https://github.com/kubernetes-sigs/gateway-api/issues/326)   |
+| Cross-Origin Resource Sharing (CORS)          | Open         | [Issue 1767](https://github.com/kubernetes-sigs/gateway-api/issues/1767) |
+| Gateway client certificate verification       | Provisional  | [GEP-91](https://gateway-api.sigs.k8s.io/geps/gep-91/)                   |
+| Authentication                                | Open         | [Issue 1494](https://github.com/kubernetes-sigs/gateway-api/issues/1494) |
+
+## Grouping the Features
+
+To reduce the number of CRDs NGINX Gateway Fabric must maintain and users have to create, we grouped the high and medium-priority features into configuration categories.
+
+For each group, we will begin implementation by focusing on the high-priority features of each group. Then, once the high-priority features are complete, we can move on to the medium-priority features.
+
+The low-priority features are out of scope for the rest of the Enhancement Proposal but may be revisited once we make progress on the higher-priority features. In addition, most features with active GEPs are not included in the groups as we want to help move the GEPs forward toward standardization instead of creating our bespoke solutions.
+
+When grouping the features, we considered the following factors:
+
+1. Use cases. Are there features that can be grouped by the use case they satisfy? For example, authentication or observability.
+2. Gateway API roles. Which features are the domain of the Cluster Operator? Which features are Application Developers responsible for?
+3. The NGINX contexts. Which contexts are the NGINX directives available in? For example, http, server, location, upstream, etc.
+
+The following picture shows the nine groups we came up with:
+
+![groups](/docs/images/nginx-functionality-groups.png)
```

Comment:
Seeing this view, I find it surprising that we have authentication, but not authorization.

Are there no NGINX features that allow/disallow traffic based on the contents of the authentication being passed in?

Reply:
authorization configuration  - this area hasn't been explored by the Gateway API, but I think it'd be better if there was a mechanism in the Gateway API (like routing rules), which could integrate with.

related thing:
it'd be great to have interoperability between auth policies and rate limiting, so that we can rate limit on per user basis.

Reply:
>Are there no NGINX features that allow/disallow traffic based on the contents of the authentication being passed in?

Like claim-based routing?

If authentication policies are applied on a per-route basis -- for example, in a filter -- aren't we effectively doing that? 

> it'd be great to have interoperability between auth policies and rate limiting, so that we can rate limit on per user basis.

@pleshakov I'll make a note of this in future work.

Reply:
> Spoke offline with @f5yacobucci, who mentioned we could support configuring allowed methods for a given path. Seems like this would be doable with the [limit_except](https://nginx.org/en/docs/http/ngx_http_core_module.html#limit_except) nginx directive.

how is that different from method-based routing like here https://github.com/nginxinc/nginx-gateway-fabric/blob/92081af949aa6c8436480563659ac8b139135d04/examples/advanced-routing/cafe-routes.yaml#L46-L49 ?

Reply:
The decision to allow/deny would be made based on the user, not the path. Some users may have read access only, while others have read/write access.

Reply:
Spoke offline, and initially, we will support passing authentication info via headers to backend applications. This will offload the authorization decision to the backend applications. We will revisit other authorization solutions at a later date.